lib/android.js

```diff
@@ -29,12 +29,22 @@ function getApi () {
       },
       '_ZN3art6Thread14CurrentFromGdbEv': ['art::Thread::CurrentFromGdb', 'pointer', []],
       '_ZNK3art6Thread13DecodeJObjectEP8_jobject': ['art::Thread::DecodeJObject', 'pointer', ['pointer', 'pointer']],
-      '_ZN3art6mirror6Object5CloneEPNS_6ThreadE': ['art::mirror::Object::Clone', 'pointer', ['pointer', 'pointer']]
     },
     variables: {
       '_ZN3art7Runtime9instance_E': function (address) {
         this.runtime_instance_ptr = address;
       }
+    },
+    extraInit: function(exports, fullAPI) {
+      fullAPI.isXposed = '_ZNK3art9OatHeader23IsXposedOatVersionValidEv' in exports;
+      let exp = exports[fullAPI.isXposed ? '_ZN3art6mirror6Object5CloneEPNS_6ThreadEm' :
+       '_ZN3art6mirror6Object5CloneEPNS_6ThreadE'];
+      if (exp !== undefined) {
+        fullAPI['art::mirror::Object::Clone'] = new NativeFunction(exp.address, 'pointer', 
+          fullAPI.isXposed ? ['pointer', 'pointer', 'int'] : ['pointer', 'pointer']);
+        return true;
+      }
+      return false;
     }
```

Comment:
I think it would be cleaner if we made use of the `optionals` support, by having two entries and adding both of them to `optionals`. We could then check if the regular one is missing, and plug in a shim function that just forwards to the Xposed one with the extra argument set to 0. (That way the implementation doesn't need to know about Xposed.)